src/datachain/query/dispatch.py

```diff
@@ -54,54 +56,49 @@ def udf_entrypoint() -> int:
     # Load UDF info from stdin
     udf_info = load(stdin.buffer)
 
-    (
-        warehouse_class,
-        warehouse_args,
-        warehouse_kwargs,
-    ) = udf_info["warehouse_clone_params"]
-    warehouse = warehouse_class(*warehouse_args, **warehouse_kwargs)
+    query: Select = udf_info["query"]
+    table: Table = udf_info["table"]
+    batching: BatchingStrategy = udf_info["batching"]
 
     # Parallel processing (faster for more CPU-heavy UDFs)
     dispatch = UDFDispatcher(
         udf_info["udf_data"],
         udf_info["catalog_init"],
         udf_info["metastore_clone_params"],
         udf_info["warehouse_clone_params"],
+        query=query,
+        table=table,
         udf_fields=udf_info["udf_fields"],
         cache=udf_info["cache"],
         is_generator=udf_info.get("is_generator", False),
+        is_batching=batching.is_batching,
     )
 
-    query = udf_info["query"]
-    batching = udf_info["batching"]
-    table = udf_info["table"]
     n_workers = udf_info["processes"]
-    udf = loads(udf_info["udf_data"])
     if n_workers is True:
-        # Use default number of CPUs (cores)
-        n_workers = None
+        n_workers = None  # Use default number of CPUs (cores)
+
+    wh_cls, wh_args, wh_kwargs = udf_info["warehouse_clone_params"]
+    warehouse: AbstractWarehouse = wh_cls(*wh_args, **wh_kwargs)
 
     with contextlib.closing(
-        batching(warehouse.dataset_select_paginated, query)
+        batching(warehouse.db.execute, query, ids_only=True)
```

Comment:
Not sure yet, but it looks like we don't need pagination here since we are only selecting IDs.
Should be tested on bigger scale and confirmed.

Reply:
our `sys__id` is 8 bytes, so on 1B scale this will take 8GB of memory by my calculation. I would still maybe leave it paginated.